mantidproject/mantid
# Implement WorkspaceObserver in place of Poco notifications in Reflectometry gui

None of the changes will be apparent when running Mantid in release, so this will have to be tested by building Mantid in Debug and running it like that.

**To Test**:
Set up your Data Directory so it looks at the ISIS Sample Dataset (if you don't have this locally you can download it from [here](http://sourceforge.net/projects/mantid/files/Sample%20Data/SampleData-ISIS.zip/download)
- Open ISIS Reflectometry (Polref) (`interfaces>Reflectometry>ISIS Reflectometry(Polref)`)
- Go to `Reflectometry` on the menu bar and `import table` 
- load `INTER_NR_test2.tbl`
- Now Go to `Reflectometry` on the menu bar and Load the table into the interface
- Confirm that no error is produced
- Try some basic workspace changes (rename, delete ,etc.) 
- Ensure that no errors are produced.

**Release Notes**:
minor change that will not affect anything in release mode, so no need to change release notes.

Fixes #14273
